pytype/tools/analyze_project/pytype_runner.py

```diff
@@ -146,19 +146,31 @@ def _module_to_output_path(mod):
 
 
 def escape_ninja_path(path: str):
-  """escape `:` in absolute path on windows."""
-  if sys.platform == 'win32':
-    new_path = ''
-    last_char = None
-    for ch in path:
-      if last_char != '$' and ch == ':':
-        new_path += '$:'
-      else:
-        new_path += ch
-      last_char = ch
-    return new_path
-  else:
-    return path
+  """Returns the path with special characters escaped.
+
+  Escape colon, dollar sign, space, and new line, for ninja
+  (as described in https://ninja-build.org/manual.html#ref_lexer).
+  This function should only ever be used once, called on a path string to turn
+  it into a ninja path string. (If you call it on a ninja path string, it will
+  render the variables inert.)
+
+  Args:
+    path: The path.
+  """
+  new_path = ''
```

Comment:
Now that there are multiple special characters involved, I'd prefer to use `re.sub` to make the code more concise. This seems to work:
```
return re.sub(r'(?P<char>[\n :$])', r'$\g<char>', path)
```
(An `import re` line is also needed at the top of the file, of course.)

Reply:
`re.sub(r'([\n :$])', r'$\1', path)` also works equivalently, and reads slightly simpler, if you prefer that.